lib/chai.d.ts

```diff
@@ -0,0 +1,1518 @@
+// <reference types="assertion-error"/>
```

Comment:
How does it deal with this type reference? Does it forces the user to install the types for `assertion-error` itself? This may pose issues due to versions mismatch between the version actually used by chai and the one uplifted to the consumer project.

Reply:
Can't we bundle an specific version of [this type definition](https://github.com/types/npm-assertion-error) to be used with Chai? This way we could always ensure it would be right (instead of depending on peer-deps).

(I don't understand much about TypeScript, so please let me know if I misunderstood what you've just said)

Reply:
The link you posted refers to the types for the _typings_ tool. This tool was used before Typescript 2, because typescript did not support types from npm. These repos are currently (slowly) maintained for older tools but the recommended way is to publish the types to the library's repository. If this is impossible, the solution is to use [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). Definitions in this repo are automatically published to the `@types` npm organization.

There are two solutions for the current issue:
- Adding type definitions to `assertion-error` (better)
- adding [`@types/assertion-error`](https://www.npmjs.com/package/@types/assertion-error) as a normal dependency to `package.json` (quicker)

Reply:
The first option seems more adequate indeed.

[As I've seen in the `assertion-error` repo](https://github.com/chaijs/assertion-error), it's not that big, so it might be a simple issue to attack.

Reply:
Oh, I had a note to deal with this and forgot about it...

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/assertion-error/index.d.ts

+1 to adding type definitions to `assertion-error` and -1 to adding any extra dependencies to `package.json`.  I'll tackle that now.

Reply:
Actually, just looking through this code, it looks like `AssertionError` is manually copy & pasted below. The comment above is `//` instead of `///` and so doesn't do anything. I'll remove it.